Redesign EditProfilePage and EditHostingPreferences

nabramow opened the pull request:
**Describe briefly what this PR is doing and why**

Redesign of the EditProfile and EditHostingPreference pages to be more organized, provide more guidance and fix some longstanding issues.

Closes #6360
Closes #6459 
Closes #6517 
Closes #6581
Closes #4143 

**Please give clear steps for how the reviewer can best test this PR**

*Please include any necessary dev environment, .env, etc. adjustments.*

- Go to EditProfile page
- Change a lot of fields
- Save
- Go to profile and make sure everything saved properly
- Try same in My Home

<!---
Checklists - you can remove one that is not applicable (ie. remove backend checklist if you only worked on the web frontend)
If you need help with any of these, please ask :)
--->
**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app
- [ x ] Added tests where relevant
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes

**Other**
*Untick the following if you'd prefer that maintainers don't push commits/merge your branch.*
- [x] A maintainer can push commits to my branch
- [x] A maintainer can merge my PR (you can also merge after approval)


<!---
Remember to request review from couchers-org/web, couchers-org/backend or an individual.
Once your code is approved, remember to merge it if you have write access
--->

nabramow committed:
Prototype of edit profile page design changes

nabramow committed:
Merge branch 'develop' into na/web/new-edit-profile-design

nabramow committed:
Remove box within a box

nabramow committed:
Improvements for mobile

nabramow committed:
Clean up for desktop status cards

nabramow committed:
More style cleanup

jesseallhands commented on app/web/features/profile/locales/en.json:
```diff
@@ -87,7 +95,16 @@
+    "preferences_subtitle": "Let others know about your hosting and meetup preferences",
+    "personal_information": "Personal Information",
+    "personal_information_subtitle": "Share details about yourself and your background",
+    "about_me_subtitle": "Tell others about yourself - this is what makes your profile unique",
```
I'm definitely doing the link wrong here, but you get the idea. We probably should force this link to open in a separate tab so they don't lose their spot on the edit profile page or trigger the warning?)

```suggestion
    "about_me_subtitle": "Tell others about yourself. This is where trust begins. Help others get a sense of who you really are. See [tips for creating a great profile](https://help.couchers.org/hc/couchersorg-help-center/articles/1725919197-how-do-i-create-a-great-profile)",
```

nabramow committed:
Change icon style and move search info box

jesseallhands commented on app/web/features/profile/locales/en.json:
```diff
@@ -87,7 +95,16 @@
-    "education": "Education"
+    "education": "Education",
+    "basic_information": "Basic Information",
+    "basic_information_subtitle": "Start with the essentials - your photo and basic details",
```
```suggestion
    "basic_information_subtitle": "Start by providing your photo, name, and location",
```
We also should add a label above the map that says "Approximate location of your home":
<img width="1441" height="868" alt="image" src="https://github.com/user-attachments/assets/fff336e4-cd1c-4c51-b50f-37624aa66c2a" />

jesseallhands reviewed (changes requested):
These are big improvements! Looks a lot more professional/modern and better organized. Most of my suggestions are related to rewording or adding a link. I like the style changes to the hosting/meet up selectors recently made. I thought maybe adding a stroke like this could help (but do what you think looks best):
https://github.com/user-attachments/assets/0c9de973-bd7f-4a52-86eb-6304857bc930

In the hosting status section, the text `I'm available to host travelers and provide accommodation` might suggest some sense of obligation to people and "provide accomodation" sounds too much like airbnb etc. This status is more for indicating eagerness (compared to just willingness), so I think we should use the following text instead:
`I am eager to let people stay with me!`

This sentence is missing "the" isn't it?: `I might be able to host, depending on circumstances`
Shouldn't it be: `I might be able to host, depending on the circumstances`?

Similarly, I think the meetup texts should be:

- Wants to meet: `I'm eager to meet up with travelers`
- Wants to meet: `I might meet up, depending on the circumstances`
- Can't meet up: `I prefer not to meet up with travelers`

The "About me" hint `This content is visible by other users when they search for you.` should read:
`The beginning of this section is visible in search results when other members search for you`

Also, the footer should be removed from this page. It's not needed on this page, and in some cases it causes users to overlook the save button (especially on high resolution desktop and on big mobile devices).

nabramow committed:
Fix link

nabramow committed:
Make location tags higher contrast

nabramow committed:
Get aboutMe length check working

nabramow committed:
Add EditProfilePage tests and get all tests passing

nabramow committed:
Fix EditHostinPreference tests

nabramow committed:
Merge branch 'develop' into na/web/new-edit-profile-design

nabramow committed:
Hide footer for edit profile pages

nabramow committed:
Merge remote-tracking branch 'origin/na/web/new-edit-profile-design' into na/web/new-edit-profile-design

nabramow committed:
Merge branch 'develop' into na/web/new-edit-profile-design

nabramow committed:
Update test

jesseallhands commented on app/web/features/profile/locales/en.json:
```diff
@@ -87,7 +95,17 @@
+    "preferences_subtitle": "Let others know about your hosting and meetup preferences",
+    "personal_information": "Personal Information",
+    "personal_information_subtitle": "Share details about yourself and your background",
+    "about_me_subtitle": "Tell others about yourself. This is where trust begins. Help others get a sense of who you really are. See <1>tips for creating a great profile</1>.",
```
Looks like the link isn't working here:
<img width="1422" height="223" alt="image" src="https://github.com/user-attachments/assets/bd859c3d-29a1-4fa5-90e8-098d5d126723" />

jesseallhands reviewed (changes requested):
Changes look good! A couple more things I found:

Regarding the below screenshot:
Instead of the MUI icon `Search` we should use `InfoOutline` instead to avoid confusion.
I couldn't find the text `Please write 150 more characters to complete your profile.` in the files, but this should probably be changed to on of the following, either:
`You must write at least 150 characters in order to interact with other people on the platform`.
OR
`**Please write at least 150 characters to unlock messaging and requests.** Genuine profiles build a community of trust. The more you share, the easier it is to connect!`

<img width="904" height="277" alt="image" src="https://github.com/user-attachments/assets/47b90bcd-df37-4886-84bd-72cef10aeb39" />


In the photo upload section, we should show the following text by default:
`**A photo of yourself is required to interact with the community.** A clear, friendly photo showing your face builds trust, helps others feel comfortable, and greatly improves your chances of receiving a response.`

I suggest we move the "Start by providing..." text above the green separator line below "Basic information", and then below the green separator line put the new photo text mentioned above.
<img width="1270" height="718" alt="image" src="https://github.com/user-attachments/assets/673dbce4-8d27-4562-90c7-cadb9a6a9858" />

nabramow committed:
Address design and wording comments

nabramow commented:
Okay @jesseallhands I addressed all of this, except I did the location for the profile photo description differently.

1) I added the info icon so it's consistent with the search instructions note

2) I put the instructions in the gray box instead of in the header. Since none of the other sections have descriptions in the header, I wanted to keep it consistent.

Also just checking you looked at the "My Home" page too as so far all your comments were about the first tab! Unless it's perfect ;-).

nabramow committed:
Add tab icons, convert to styled

nabramow committed:
Adjust test

jesseallhands commented:
>Also just checking you looked at the "My Home" page too as so far all your comments were about the first tab!

Oops yeah I need to look at that still! 😅

jesseallhands commented on app/web/features/profile/locales/en.json:
```diff
@@ -65,15 +64,23 @@
+    "household_habits": "Household Habits",
+    "hosting_preferences_subtitle": "Let guests know what you're comfortable with",
+    "about_home_subtitle": "Tell guests about your home and what they can expect",
+    "household_details_subtitle": "Share information about who lives in your home",
```
```suggestion
    "household_details_subtitle": "Share more details about your household",
```

nabramow replied:
Done

jesseallhands reviewed (changes requested):
Sorry for overlooking the "My home" section the first time around! Here are the things I found on that page:

1. If you empty the "max number of guests" field, and click submit, you get an "assertion failed error:
<img width="600" height="649" alt="image" src="https://github.com/user-attachments/assets/bb102ca8-f665-4653-b381-ef32f2c672ad" />This field should probably just be a dropdown menu. This works well because the vast majority of the numbers selected will be between 1-5, and it will be rare that someone would enter a number >10. Alternatively, a number input toggle (like [this](https://v6.mui.com/base-ui/react-number-input/#minimum-and-maximum) MUI component) could work since the vast majority of inputs will be 1-5.

2. Some fields that should require the user to actively make a selection currently seem to default to a selection. For example, "sleeping privacy" defaults to "private space". Furthermore "ask me" was meant to be a placeholder only for "null" values leftover from the alpha version. "ask me" shouldn't be an option we offer to new users.
Here are the fields that shouldn't have a default (user should be forced to make a selection): - Okay with smoking (this is a quite important factor for health/comfort)
- Sleeping privacy
- Parking details (if parking available checkbox is activated)
These required fields will need a modal error message if the user tries to save when they are empty. The modal should say: `Oops! You missed some required fields. Please complete them before saving.` or `Please fill in all required fields before saving.`

3. The following options should be displayed differently to avoid ambiguity (eg. the children part, _legal yikes_ 😅):
- `Hosts children` --> `Hosts people with children`
- `Hosts pets` --> `Hosts people with pets`
- `Okay with drinking` -->  `Okay with people drinking`

3. The `Let guests know what you're comfortable with` section has some options that don't really fit with the rest of the options. For example:
- Wheelchair accessible
- Tent camping available

These two options have nothing to do with what you're comfortable with - but instead what your home facilities are able to accommodate. Therefore, the following options should be moved to the `Household Details` section, and probably in the following order _**after**_ the `Household Members` subsection:
- Wheelchair accessible
- Tent camping available
- Parking available
  - Parking details (only display if `parking available` is activated)

nabramow committed:
Fix merge conflicts and merge develop

nabramow committed:
Address MyHome comments

nabramow commented:
> Sorry for overlooking the "My home" section the first time around! Here are the things I found on that page:
> 
> 1. If you empty the "max number of guests" field, and click submit, you get an "assertion failed error:
Fixed!

> This field should probably just be a dropdown menu. This works well because the vast majority of the numbers selected will be between 1-5, and it will be rare that someone would enter a number >10. Alternatively, a number input toggle (like [this](https://v6.mui.com/base-ui/react-number-input/#minimum-and-maximum) MUI component) could work since the vast majority of inputs will be 1-5.
MUI Base is actually a different package than what we use so I don't have access to that.  The current one we use is a number input, with a min of 0 and a max of 10. You can click up and down arrows to increase and decrease the number.

> 2. Some fields that should require the user to actively make a selection currently seem to default to a selection. For example, "sleeping privacy" defaults to "private space". Furthermore "ask me" was meant to be a placeholder only for "null" values leftover from the alpha version. "ask me" shouldn't be an option we offer to new users.

Fixed.
>    Here are the fields that shouldn't have a default (user should be forced to make a selection): - Okay with smoking (this is a quite important factor for health/comfort)
> 
> * Sleeping privacy
> * Parking details (if parking available checkbox is activated)

Fixed.

>   These required fields will need a modal error message if the user tries to save when they are empty. The modal should say: `Oops! You missed some required fields. Please complete them before saving.` or `Please fill in all required fields before saving.`

I tried to do this, but it's actually quite difficult as we made a custom select and our empty value is not null on the backend. So it revolves re-writing a lot of the logic, thus I will leave it for now to not explode this PR. We can make a separate ticket if we want to come back to this.

> 3. The following options should be displayed differently to avoid ambiguity (eg. the children part, _legal yikes_ 😅):
> 
> * `Hosts children` --> `Hosts people with children`
> * `Hosts pets` --> `Hosts people with pets`
> * `Okay with drinking` -->  `Okay with people drinking`

Done.

> 3. The `Let guests know what you're comfortable with` section has some options that don't really fit with the rest of the options. For example:
> 
> * Wheelchair accessible
> * Tent camping available
> 
> These two options have nothing to do with what you're comfortable with - but instead what your home facilities are able to accommodate. Therefore, the following options should be moved to the `Household Details` section, and probably in the following order _**after**_ the `Household Members` subsection:
> 
> * Wheelchair accessible
> * Tent camping available
> * Parking available
>   
>   * Parking details (only display if `parking available` is activated)

Done

nabramow committed:
Fix type issues and generate protos to fix build

nabramow committed:
Fix test

nabramow committed:
Merge branch 'develop' into na/web/new-edit-profile-design

nabramow committed:
Simplify aboutMe logic

nabramow committed:
Fix tab spacing

jesseallhands reviewed (changes requested):
Everything looks good, except I found 2 things on the "my home" page, both I was able to replicate on FF and chrome:

1. Even if the number of hosts is populated, it also shows the helper text in the box simultaneously:
![Screenshot_20250810_012816_Firefox](https://github.com/user-attachments/assets/f6e520a4-4995-481f-87d3-7ffd4ec64818)

2. The save button sometimes doesn't appear on the "my home" page, and I think it is related to users who don't select a sleeping privacy or some other fields. It's also a bit problematic that you can set your hosting status to "can host" and also omit information very important to hosting like sleeping privacy, max number of guests, smoking, etc.

Here is a video that shows the save button not appearing:

https://github.com/user-attachments/assets/fd055a14-5894-4b93-b060-c908d55f9476

nabramow committed:
Merge branch 'develop' into na/web/new-edit-profile-design

nabramow commented:
> 1. Even if the number of hosts is populated, it also shows the helper text in the box simultaneously:

Are you sure you're on the most recent Vercel preview of this? I'm not able to replicate it, see video:

https://github.com/user-attachments/assets/f0b862dc-e736-4972-bfbd-e71cdf743fda

Or can you tell me exactly the steps to make this happen?

> Here is a video that shows the save button not appearing:

@jesseallhands Did you change anything? The save button won't show until something changes from the existing values. Can you tell me what steps you did to make this happen? I'm also not able to replicate this.

> It's also a bit problematic that you can set your hosting status to "can host" and also omit information very important to hosting like sleeping privacy, max number of guests, smoking, etc.

This is outside the scope of this PR.

jesseallhands reviewed (dismissed):
I can't replicate the overlapping text issue anymore, so that's all good!

It appears the missing save button issue only happens when you don't change anything. In my testing I changed stuff and then changed it back, which is why I was confused. If it's easy to just make the save button always visible even if nothing has changed, we should do that as it's pretty common behavior (unless we wanna change the button text to "save changes").

nabramow committed:
Make tab style more specific to EditProfile

nabramow committed:
Change save button text to save changes

nabramow commented:
> I can't replicate the overlapping text issue anymore, so that's all good!

Sweeet.

> It appears the missing save button issue only happens when you don't change anything. In my testing I changed stuff and then changed it back, which is why I was confused. If it's easy to just make the save button always visible even if nothing has changed, we should do that as it's pretty common behavior (unless we wanna change the button text to "save changes").

I actually purposely changed it to only show when something is changed as I found it quite confusing the other way! But I just changed the text to "save changes".

nabramow committed:
Update test translation string

nabramow committed:
Fix merge conflicts, checkboxes and add test

jesseallhands reviewed (approved):
LGTM!

nabramow committed:
Merge branch 'develop' into na/web/new-edit-profile-design
